message/router.go

```diff
@@ -99,10 +99,22 @@ func NewRouter(config RouterConfig, logger watermill.LoggerAdapter) (*Router, er
 	}, nil
 }
 
+type Middleware struct {
+	Handler     HandlerMiddleware
+	HandlerName string
+}
+
+func NewMiddleware(handler HandlerMiddleware, handlerName string) Middleware {
```

Comment:
This function probably doesn't have to be exported, as it's used only internally by the router.

Reply:
Actually, the `Middleware` struct can be private as well. It could be confusing to the end user why there's public `Middleware` and `HandlerMiddleware`.